pkg/observability/tracing.go

```diff
@@ -0,0 +1,62 @@
+package observability
+
+import (
+	"context"
+	"log"
+
+	"github.com/gobuffalo/buffalo"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+// ObserverContext is a context that wraps around a buffaloCtx and also keeps a span context
+// for spans to use it
+type ObserverContext struct {
+	buffalo.Context
+	SpanCtx context.Context
```

Comment:
it seems that `SpanCtx` field is not being anywhere. This is the field that actually carries the span tree info..therefore, if you pass `ObserveContext` as a whole and not this field in particular, OpenCensus won't recognize the parents (https://github.com/census-instrumentation/opencensus-go/blob/master/trace/trace.go#L162) 

One solution is to have our StartSpan function that does what the trace.StartSpan does, but we can't do what StartSpan does because they use unexported fields. 

Another solution is to implement your own context.Context interface on top of ObserveContext and you'd basically call both buffalo.Context and SpanCtx methods and return whichever is not nil. But that seems ugly at first. 

Maybe we should talk to the OpenCensus peeps (cc @tbpg ) and see what's a good solution to have Spans spun inside a Buffalo.Context and then return a BuffaloContext and not a context.Context.

Reply:
I am thinking of type asserting to buffalo.Context the context returned by  StartSpan when called StartBuffaloSpan and in the middleware and getting rid of the ObserverContext. 

Thoughts?